packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2466,6 +2491,382 @@ class _MenuDirectionalFocusAction extends DirectionalFocusAction {
+/// The arguments to the function are as follows:
+///
+/// * The `context` supplies the [BuildContext] to use.
+/// * The `label` is the [MenuAcceleratorLabel.label] attribute for the relevant
+///   [MenuAcceleratorLabel] with the accelerator markers stripped out of it.
+/// * The `index` is the index of the accelerator character within the `label`
+///   that applies to this accelerator. If it is -1, then the accelerator should
+///   not be highlighted. Otherwise, the given character should be highlighted
+///   somehow in the rendered label (typically with an underscore).
+/// {@endtemplate}
+///
+/// See also:
+///
+/// * [MenuAcceleratorLabel.defaultLabelBuilder], which is the implementation
+///   used as the default value for [MenuAcceleratorLabel.builder].
+typedef MenuAcceleratorChildBuilder = Widget Function(
+  BuildContext context,
+  String label,
+  int index,
+);
+
+/// A widget that draws the label text for a menu item (typically a
+/// [MenuItemButton] or [SubmenuButton]) and renders its child with information
+/// about the currently active keyboard accelerator.
+///
+/// On platforms other than macOS and iOS, this widget listens for the Alt key
+/// to be pressed, and when it is down, will update the label by calling the
+/// builder again with the position of the accelerator in the label string.
+/// While the Alt key is pressed, it registers a shortcut with the
+/// [ShortcutRegistry] mapped to a [VoidCallbackIntent] containing the callback
+/// defined by the nearest [MenuAcceleratorCallbackBinding].
+///
+/// Because accelerators aren't used on macOS and iOS, the label ignores the Alt
+/// key on those platforms, and the [builder] is always given -1 as an
+/// accelerator index. Accelerator labels are still stripped of their
+/// accelerator markers.
+///
+/// The built-in menu items [MenuItemButton] and [SubmenuButton] already provide
+/// the appropriate [MenuAcceleratorCallbackBinding], so unless you are creating
+/// your own custom menu item type that takes a [MenuAcceleratorLabel], it is
+/// not necessary to provide one.
+///
+/// {@template flutter.material.MenuAcceleratorLabel.accelerator_sample}
+/// {@tool dartpad} This example shows a [MenuBar] that handles keyboard
+/// accelerators using [MenuAcceleratorLabel]. To use the accelerators, press
+/// the Alt key to see which letters are underlined in the menu bar, and then
+/// press the appropriate letter. Accelerators are not supported on macOS or iOS
+/// since those platforms don't support them natively, so this demo will only
+/// show a regular Material menu bar on those platforms.
+///
+/// ** See code in examples/api/lib/material/menu_anchor/menu_accelerator_label.0.dart **
+/// {@end-tool}
+/// {@endtemplate}
+class MenuAcceleratorLabel extends StatefulWidget {
+  /// Creates a const [MenuAcceleratorLabel].
+  ///
+  /// The [label] parameter is required.
+  const MenuAcceleratorLabel(
+    this.label, {
```

Comment:
Can/should anything be validated about the label, or is there no invalid state?  Like I guess a label without an `&` will just not underline anything?